add access key option for azure_rm_storageblob

mmyyrroonn opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Add access key option for azure_rm_storageblob. It can making requests to the storage account in other subscription with access key. Fixes #55350 
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
azure_rm_storageblob

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

Fred-sun commented:
@yungezz @yuwzho @zikalino Please review this PR when you're available! Thanks a lot!

yungezz commented on lib/ansible/modules/cloud/azure/azure_rm_storageblob.py:
```diff
@@ -111,6 +111,11 @@
         choices:
             - container
             - blob
+    access_key:
```
shared_access_key

yungezz commented on lib/ansible/modules/cloud/azure/azure_rm_storageblob.py:
```diff
@@ -111,6 +111,11 @@
             - blob
+    access_key:
+        description:
+            - Use access key to authenticate your applications.
```
can we add function in storage module to generate sas?

yuwzho commented on lib/ansible/modules/cloud/azure/azure_rm_storageblob.py:
```diff
@@ -81,7 +81,7 @@
         description:
             - Name of the resource group to use.
-        required: true
+            - If C(resource_group) is not provided, the field C(shared_access_key) must be provided.
```
Required when I(shared_access_key) not defined.

yuwzho replied:
https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#linking-within-module-documentation

mmyyrroonn replied:
done

yuwzho commented on lib/ansible/modules/cloud/azure/azure_rm_storageblob.py:
```diff
@@ -111,6 +111,11 @@
+    shared_access_key:
+        description:
+            - Use shared access key to authenticate your applications.
+            - Required when making requests to the storage account in other subscription.
```
Required when I(resource_group) not defined

mmyyrroonn replied:
done

yuwzho commented on lib/ansible/modules/cloud/azure/azure_rm_storageblob.py:
```diff
@@ -81,7 +81,7 @@
     resource_group:
         description:
             - Name of the resource group to use.
```
- Resource group name of the storage account.
- If the storage account is in another subscription rather than `AZURE_SUBSCRIPTION_ID`, use I(shared_access_key) to access it.

mmyyrroonn replied:
done

yuwzho commented on lib/ansible/module_utils/azure_rm_common.py:
```diff
@@ -537,12 +537,16 @@ def check_provisioning_state(self, azure_object, requested_state='present'):
-            # Get keys from the storage account
             self.log('Getting keys')
-            account_keys = self.storage_client.storage_accounts.list_keys(resource_group_name, storage_account_name)
+            if provided_access_key is not None:
```
`if provided_access_key:  ## avoid empty string`

mmyyrroonn replied:
done. Thanks!

yuwzho commented:
shipit

sky-joker committed:
Added zabbix module zabbix_user_info

[update] Fix zabbix_user_info module to use missing_required_lib helper and add zabbix logout processing.

Fixed the following issue.

* https://github.com/ansible/ansible/pull/58002
* https://github.com/ansible/ansible/issues/58459

change version_added 2.9 to 2.10

Fred-sun commented:
@MyronFanQiu  Please help fix the shippable error when you're free! Thank you very much!

Fred-sun commented:
@yuwzho @MyronFanQiu Can we merged this PR or waiting others to review? Please check! Thanks a lot!

Fred-sun commented:
ready_for_review

Fred-sun commented:
@yungezz @nitzmahone Please take a review this when you're free! Thank you very much!

yungezz commented:
rebuild_merge

mariolenz committed:
VMware: Fix issue with order of changes in vmware_vcenter_statistics

goneri committed:
ansible-test: vcenter+Worldstream: hide password

Hide the temporary password when `ansible-test` is called with the
`--redact` parameter.

abadger committed:
Update the rpm spec file to include ansible-test

* Include ansible-test in a subpackage
* Sync with the ansible engine spec file
* Addresses https://github.com/ansible/ansible/issues/60380 to the
  extent that we're likely to fix that.

leroyguillaume committed:
add option to define keysize at luks partition creation (#61522)

* add option to define keysize at luks partition creation

* Bump version_added.

* Extend tests.

pabelanger committed:
Remove unused import for iosxr l3_interfaces (#61563)

This was caught by our ansible network collections migration script.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

mmazur committed:
kubevirt: also add wait_sleep (#61020)

carbolymer committed:
#59096 Fix for not working azure_rm_aks (#59097)

Fred-sun committed:
Set the default value for the managed_disk_type (#60749)

MaxBab committed:
os_server - add "tag" to instance nics (#61119)

A custom "tag" could be passed to the instance metadata with the nics.
Add support for the "tag" to the module.

gundalow committed:
Disable Automerge (#61674)

A few PRs have been automerged since devel became 2.10 even though they still had `version_added: 2.9`

Lets disable automerge till:
a) We find a proper fix (automerge should so `rebuild_merge`?)
b) We wait 7 days since devel became 1.10 to cause stale_ci

mkrizek committed:
perf: reduce getting handler variables (#59969)

* perf: reduce getting handler variables

ci_complete

* Do not re-create templar

fetzerch committed:
influxdb: Add support for 'path' parameter (#61173)

While it's possible to host InfluxDB on a different path prefix
(through a reverse proxy such as nginx or traefik), the Ansible
influxdb_* modules did't support that use case.

The patch adds a 'path' parameter, which is passed to the influxdb
Python client.

Example for connecting to InfluxDB running on http://localhost/influxdb

    - hosts: all
      tasks:
        - influxdb_database:
            database_name: test
            path: influxdb
            port: 80

vitalii-kostenko committed:
Ansible FTD Module improvements and tests update. (#60640)

* Ansible FTD Module improvements and tests update.

* Fix sanity tests

* Fix formatting

mulatinho committed:
luks_device.py: Allow manipulate LUKS containers with label or UUID (#61603)

* luks_device.py: Allow manipulate LUKS containers with label or UUID

- Allow create a LUKS2 container format with label support
- Allow manipulate (open, close, modify) an LUKS container based on
  both label (LUKS2 format) or UUID instead of using devices only.

Fixes: #58973
Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

* test_luks_device.py: organizing tests to support labels

- Add label on some tests and fix errors reported by Shippable

Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

* luks_device.py: adjusting versions and messages

- Modifying version_added from 2.9 to 2.10
- Fixing some messages
- Created a changelog fragment
- Moving blkid from scope

Fixes #58973
Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

Akasurde committed:
VMware: Fix get_tags_for_vm API in vmware_vm_info module (#61697)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

dgalpaj committed:
feat: support postgres 11 in azure_rm_postgresqlserver (#58727)

jotai73 committed:
Update homebrew_cask example docs (#61637)

Changed homebrew_cask module flag from "allow_external_apps" to "accept_external_apps" in example section

Qalthos committed:
Avoid infinite recursion on login (#61617)

willthames committed:
Remove willthames from AWS and Openshift notifications (#61670)

I no longer desire to be mentioned on most AWS changes.
If there are modules I'm on the authors list then I'll remain
notified.

Also remove me from openshift and kubectl plugins, I don't use
those.

Commit:
Fix network resource module wording and typographical errors (#61613)

liquidat committed:
Rename checkpoint to check_point due to legal reasons, fix #61145 (#61172)

* Fix #61145

* Fix type keyword

* Removed file fragment

t0mk committed:
Fix missing pass of IPXe params to API client call, fixes #32910 (#32912)

simon04 committed:
Fix typesetting of "GitLab" (#61702)

tahir-gvs committed:
Clarity on mutual exclusion of Username and Token (#61668)

##### SUMMARY
Although there is an example showing that username is not required with personal access token, it will be nice to more clearly state in options description. 
##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

it-praktyk committed:
Remove unnecessary information (#61657)

sestrella committed:
Update unit tests requirements/units.txt path (#61656)

resmo committed:
botmeta: unmaintain resmo from cloudstack (#61561)

Yannis100 committed:
net_user documentation typo configured_password (#61556)

password -> configured_password in the last example

amenonsen committed:
Remove duplicated word

We print "%s success", so there's too much success going on.

kbreit committed:
Docs - Add Meraki to platform option list (#61651)

it-praktyk committed:
Correct indentation of the code in the example (#61622)

ymartin59 committed:
Fix examples for bigip_file_copy (#61686)

wimnat committed:
Fix required_together statement - access_logs_s3_prefix is not required as per the documentation (#52017)

anshulbehl committed:
adding a collection local try section (#61626)

anryko committed:
datadog_monitor: Add process alert support (#60325)

theo-o committed:
grafana_plugin: update documentation (#60600)

samdoran committed:
Rebalance Azure tests groups

samdoran committed:
Increase unit test timeout to account for long image download times (#61717)

cristidas committed:
Ecs taskdefinition secrets (#60112)

* adding specific examples for using environment and secrets in the container definition

gundalow committed:
version_added: '2.10' for post branch commits (#61678)

Due to bot automerge a few PRs were merged before version_added had been updated.

https://github.com/ansible/ansible/pull/61674

ETNyx committed:
docs: Update apt_key.py add requirements of gpg (#61552)

* docs: Update apt_key.py add requirements of gpg. The module needs to have gpg installed, otherwise it will end up with error {"changed": false, "msg": "Failed to find required executable gpg in paths: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"

Co-Authored-By: John R Barker <john@johnrbarker.com>

yuji-k64613 committed:
Fixes #59613 (#60041)

Andersson007 committed:
postgresql: Move PgMembership to module_utils (#61598)

* postgresql: move PgMembership to module_utils

* postgresql: move PgMembership to module_utils, optimize logic

Commit:
Issue #41447 - support AWS Aurora S3 privileges in mysql_user module (#59663)

mnecas committed:
ovirt search name with space (#59184)

* init ovirt_search_name_with_space

* add search_by_name with whitespace

* add comments

nikkytub committed:
Removed additional fetch from config context because it is now available in the host from Netbox v2.6 onwards (#59028)

* Removed additional fetch and config context as an option as it is now available in the host from Netbox v2.6

* Added config-context as an option

catcombo committed:
Fix Python 2.7 compat (#58816)

Commit:
Add example of "parent_group" usage in aws_ec2.py (#60430)

##### SUMMARY
Looks like the "parent_group" option in "keyed_groups" is a very recent feature that is not being documented. 
This pull request just adds a simple example of the usage of this useful feature.



##### ISSUE TYPE
- Docs Pull Request

+label: docsite_pr

migumun committed:
Fix Decode Bug in Py3 (#61691)

* Fix Decode Bug in Py3

* Update fortios.py

* Remove special char from name

* Ensure full compatibility when ansible is install from Py2

geerlingguy committed:
Fix documentation to reflect CLI usage for collections-path (#60164)

goneri committed:
debian: prepare the ansible-test package

- Generate an extra package for `ansible-test`.
- Manually install the files because CDBS get confused by our two
  Python packages
- Refresh the package descriptions to match what is done with the RPM

opoplawski committed:
osx_defaults - doc tweaks (#60881)

* [osx_default] We are looking for the "defaults" executable, not "osx_defaults"

* [osx_default] Add example for setting system wide default

movinalot committed:
Add ucs_query (#60607)

* Add ucs_query

* fix sanity erros

* Change version_added to 2.10

billdodd committed:
Fix KeyError exceptions in GetLogs command (#59877)

* fix KeyError exceptions in GetLogs command

* add changelog fragment

* refine the list of LogEntry properties to collect

billdodd committed:
Fix issues and limitations in account mgmt commands (#58441)

* fix issues and limitations in account mgmt commands

* fix pep8 and example yaml errors

* remove new option and update option aliases

* find next empty slot when adding user via PATCH

jtyr committed:
Read update file as utf-8 (#55253)

jborean93 committed:
Set win_pagefile as unstable as thats what it is (#61732)

caphrim007 committed:
Migrate ec2_eip module to boto3 (#61575)

* Migrate ec2_eip module to boto3

This patch is a step towards the integration of several PRs that have
attempted to migrate this code

closes #55190
closes #45478

Follow-up PRs will address the outstanding changes made in #55190

mches committed:
Fix creation of DigitalOcean droplets using digital_ocean_droplet module (#61655)

* Omit disallowed 'id' attribute in request body to create a droplet

* Add changelog fragment for PR #61655

Fred-sun commented:
review_need

justjais committed:
Fixes IOS L3 intermittent zuul failure (#61682)

* fix ios l3 intermittent failure

* fix self

* dict to ordered dict

* fix diff_again fn

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* remove orderdict as its expensive

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

* update verify fn

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

trishnaguha committed:
Remove choices from gather_network_resources facts and fix negated all, min (#61362)

* remove choices from gather_network_resources facts and allow negating subset without needing to add a new subset specific for negation

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

* negated all, min should not return any fact

Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

florianpaulhoberg committed:
add trustcacerts option for JAVA keytool (#61750)

* add trustcacerts option for JAVA keytool

* fixed whitespace

* added type in documentation

s-hertel committed:
aws_s3 - wait for the bucket before setting ACLs (#61735)

* Wait for the bucket to become available if possible before setting ACLs

s-hertel committed:
aws_codebuild tests - add retries and delay instead of a pause task (#61731)

- Add retries instead of a pause task
- Shorten the IAM role name length
- Put the IAM role name in defaults/main.yml instead
- Fix the aws_codepipeline tests too

NilashishC committed:
Make *_bgp modules collection safe (#61761)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

kustodian committed:
Fix systemd start state with deactivating service state

gundalow committed:
aluvenko isn't reviewing anymore (#61764)

* aluvenko isn't reviewing anymore

aluvenko hasn't been around for a while, so remove them

* Update BOTMETA.yml

samccann committed:
Minify theme (#61734)

* moved most ansible edits out of theme.css
* remove unnecessary edit to theme.css
* replace with upstream minified theme.css for 0.4.3

kucharskim committed:
Allow 13 asterisk characters in password field without warning (#54893)

On OpenBSD, 13 asterisk characters as a password hash, marks the
account as disabled. Otherwise daily(8) script which executes
security(8) will email operator about not properly locked accounts.

Before the diff, we see following warning:

> [WARNING]: The input password appears not to have been hashed. The 'password' argument must be encrypted for this module to work properly.

After the diff, warning is gone.

s-hertel committed:
codebuild/codepipeline - fix regex for shippable resource_prefix (#61783)

* Fix bad assumption about shippable resource_prefix for codebuild and codepipeline tests
* Update test/integration/targets/aws_codepipeline/defaults/main.yml

sluther committed:
fix erroneous failures in docker_compose due to deprecation warnings … (#61650)

* fix erroneous failures in docker_compose due to deprecation warnings from docker (#60961)

* Update error handling to work with new method of capturing output

Co-Authored-By: Felix Fontein <felix@fontein.de>

* update error handling

* fix syntax error

* fix indentation

* fix indentation (again)

* remove erroneous line

pabelanger committed:
Remove unused import for vyos facts (#61784)

This is no longer needed and can be removed.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

samccann committed:
Minify ansible (#61792)

* make all comments special to survive minify
* minified css

pabelanger committed:
Remove unused import for cisco ios facts (#61790)

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

swartzn committed:
Fix documentation and facts_from_proxy value in the netapp_e_facts module. (#61072)

s-hertel committed:
s3 - improve waiting for the bucket (#61802)

gundalow committed:
Proposals aren't really used much, delete it (#61800)

pabelanger committed:
Remove unsed import for junos facts (#61787)

This is no longer needed.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

tremble committed:
ec2_launch_template: Fix integration tests (#61260)

* ec2_launch_template: (integration tests) make sure security_token is optional

* ec2_launch_template: (integration tests) add dependencies at the top level so they're pulled into the docker containers

* Update Hacking Compute Policies for Launch Templates

sivel committed:
Don't truncate the last character from galaxy urls. Fixes #61624 (#61775)

pabelanger committed:
Remove unsed import for eos facts module (#61795)

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

jborean93 committed:
ansible-galaxy fix --server option for roles (#61820)

pabelanger committed:
Remove unused import from iosxr facts (#61785)

This is no longer needed and can be removed.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

Akasurde committed:
ovirt: Fix API calls in ovirt_storage_domain (#61825)

* Fixed argument in _attached_sds_service API
* Fixed argument in _maintenance API

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

billdodd committed:
Fix power command ResetType mapping logic (#59927)

* fix power command ResetType mapping logic

* add changelog fragment

felixfontein committed:
Azure _info modules: fix returned ansible_facts when called as _info (#61805)

* Fix Azure _info modules which still returned ansible_facts.

* Adjust PR #.

* Fix wrong assignment.

* Reorganize code for clarity.

Akasurde committed:
gitlab_hook: Update documentation (#61837)

* Update docs for note_events

Fixes: #61833

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

fabianvf committed:
Fix bug preventing wait_condition from being respected when using apply (#61493)

Qalthos committed:
Unify ios_logging documentation with accepted options (#61777)

* Unify ios_logging documentation with accepted options

* Apply suggestions from code review

Co-Authored-By: Alicia Cozine <879121+acozine@users.noreply.github.com>

* Copy suggestion over to eos_logging

ndclt committed:
documentation: modify the link to the docker.txt listing all images (#61838)

The file was moved.

jborean93 committed:
Fix galaxy server reference in the docs (#61823)

goneri committed:
test: disable the docker swarm tests (#61816)

`docker_swarm` tests tend to hang during the creation of the Swarn
cluster.

See: https://github.com/ansible/ansible/issues/61815

goneri committed:
azure_rm_iothub: disable the functional test (#61854)

The test fails with: Operation failed with status: 'Unauthorized'
This commit disable the test until the situation is resolved.

See: https://github.com/ansible/ansible/issues/61852

js-max committed:
Update hashi_vault.py documentation  with kv V2 example (#61221)

evgeni committed:
don't set default:None in documentation when there is no default (#61111)

otherwise "None" will be rendered as a string on docsite

gen2kes committed:
VMware: Add disk_mode parameter in vmware_guest_disk (#60406)

* Solve issue 58592

Added support to select disk_mode.

travisnburton committed:
Update mso_schema_template_deploy.py (#61849)

module names are incorrect in EXAMPLES:  changed module names from 'mso_schema_template' to 'mso_schema_template_deploy'

abadger committed:
Update release name for 2.10

Add a list of previously used release names to make it easy to tell what
release names are no longer usable.

Add a test that new release names have been added to the used list.

Fixes #61616

Qalthos committed:
Connect before reading the prompt? (#61797)

Yannis100 committed:
Typo 'state:' net_lldp (#61543)

'state:' should be "absent" not "lldp"

lcts committed:
Add caution about handlers & import to Pitfalls (#61060)

* Add caution about handlers & import to Pitfalls on the playbooks_reuse page.
The fact that handlers lose their name: when using import_tasks:, while logical, is not intuitive and should be noted.
Co-Authored-By: Sandra McCann <samccann@redhat.com>

justjais committed:
To fix IOSXR L3 Interfaces idempotency failures (#61860)

* fix zuul idemptency failures

* adding eol

Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

kbreit committed:
meraki_organization - Add warning about organization deletion (#61886)

* meraki_organization - Add warning about organization deletion
The documentation is now more explicit about the ramifications of using `state: absent` in a task.

abadger committed:
Fix rpm dependencies for ansible-test

Needs to require ansible = version rather than ansible-version

jillr committed:
Add tests for new alias deprecation functionality (#61476)

* Add tests for the alias deprecation added in #61245

tremble committed:
ec2_vpc_subnet test cleanup (#61663)

* aws_vpc_subnet: (integration tests) migrate to module_defaults

* aws_vpc_subnet: (integration tests) remove hard coded assumption that AZ A exists.

While Amazon now tends to enable all AZs in a region, new customers in us-west-1 are only assigned 2 out of the 3 AZs, which might not include AZ a

* ec2_vpc_subnet: (integration tests) General cleanup

- use "is changed" rather than .changed
- clean up labelling of a couple of assertions (C&P fail)

tremble committed:
Fixup iam_group integration tests and return value documentation (#61243)

* iam_group: (integration tests) migrate tests to module_defaults

* iam_group: (integration tests) migrate to using temporary user and group with {{ resource_prefix }}

* iam_group: (integration tests) fix test, checking the return values

* iam_group: (integration tests) Add some more tests around the behaviour of 'changed'

* iam_group: (docs) Update documentation of iam_group return value

* Update AWS testing policies to enable group/user management

nitzmahone committed:
allow external collections under the ansible NS (#61908)

* fixes #59988

felyl-github committed:
Update os_security_group_rule.py (#61262)

Currently, this module does not support to add 132 SCTP rule in security group rule of openstack. Could we add this support in this module? Thanks~

ganeshrn committed:
Fix ansible-connection persist after playbook run complete issue (#61591)

* Fix ansible-connection persist after playbook run issue

*  PR https://github.com/ansible/ansible/pull/59153 to add support
   for delaying the ansible-connection added an old issue of
   ansible-connection persisting even after playbook run is finished
   till either command timeout or connect timeout is triggered.
   ansible-connection persist after playbook execution is done
   and also delays the connection initilization untill a method
   in invoked from module side on the connection object.

* Add chanegelog

Andersson007 committed:
Fix mysql.py logic related to 53326 (#61832)

samdoran committed:
User - correct group removal description in docs (#61936)

sivel committed:
Properly hide limit and list-hosts args from ansible-inventory (#61618)

* Properly hide limit and list-hosts args from ansible-inventory. Fixes #61604

* Add changelog fragment

* Consolidate limit

* Fix positional argument with --graph

* Properly error for hidden arguments

* linting issue

* host pattern changelog

webknjaz committed:
Update the number of contributors in README: 4600+ (#61949)

tremble committed:
aws_asg: Fix idempotency when using tags and metrics (#61284)

* Update AWS hacking policy to enable ASG Tagging management

* aws_asg: Add tests for ASG Tagging (including idempotency)

* aws_asg: ignore sort order when comparing tags on the ASG (fix idempotency)

* ec2_asg: (integration tests) test for idempotency when managing metrics collection

* ec2_asg: sort list of enabled metrics to ensure clean comparisons.

acalm committed:
os_port allowed_address_pairs and extra_dhcp_opts list of dicts comparison fix (#56577)

* compare list of dicts

* update example for dhcp_server_opts to include ip_version which is automatically added by openstack

* add note about dhcp_server_opts

* add changelog fragment

* fix forgotten exception+pass

* no need to excplicitly check for None

* fix oops

* fix import error

* missed missing_required_lib

* changelog fragment formatting and grammar fixes

* update requirements in documentation and fix spelling

ctrufan committed:
Fixes to ecs_certificate cert chain for #61738 (#61858)

* Fixes to ecs_certificate cert chain for #61738

* Added changelog fragment

* Fixes to ecs_certificate for cleaner join, and better integration test

* Fix integration test formatting

* End cert chain with a \n

* Update changelogs/fragments/61738-ecs-certificate-invalid-chain.yaml

Co-Authored-By: Felix Fontein <felix@fontein.de>

* Update main.yml

felixfontein committed:
docker_container: improve port range parsing error behavior (#61740)

* Improve port range parsing.

* Add changelog.

chkp-amiads committed:
test_cp_mgmt_address_range module (#61867)

* test_cp_mgmt_address_range module

* fix test issues

- file location
- api_call is missing from facts
- metclass and import items

pabelanger committed:
Clean up unused imports for nxos related code (#61856)

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

Akasurde committed:
Remove unnecessary pass statement (#61749)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sdodsley committed:
purefa_info: Fix failure due to API version check (#61925)

Akasurde committed:
atomic_image: Fix absent state (#61827)

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

equelin committed:
VMware: Add new module VMware vSAN health infos (#58636)

ssato committed:
Allow junos_config changes the candidate configuration only (#61969)

This allows junos_config to changes the candidate configuration only and
does not commit it as the active configuration at once w/ the
'check_commit' option.

Commit:
fix: adds prefix length error to eos cli plugin (#61747)

yuandongx committed:
for capability/exchange/1.0 (#60630)

* for capability/exchange/1.0

* for capability/exchange/1.0

* update for shippable

* Update ce.py

* Update ce.py

sivel committed:
Add collections ticket stub (#62001)

* Add collections ticket stub

* typo

Co-Authored-By: John R Barker <john@johnrbarker.com>

Qalthos committed:
Debug location of ansible-connection when found (#61998)

Qalthos committed:
Add missed eos modules to porting guide (#61994)

tremble committed:
Fix behaviour of module_utils/ec2 compare_policies when dealing with bare bools and ints. (#61115)

* module_utils/ec2: (unit tests) Move unit tests for module_utils/ec2.py into test/units/module_utils

- compare_policies was refactored from s3_bucket
- "ec2_utils" doesn't seem to have ever existed

* module_utils/ec2: (unit tests) Add unit test for comparing quoted and unquoted bools and numbers within policies

As per https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html

"Values are enclosed in quotation marks. Quotation marks are optional for numeric
and Boolean values."

* module_utils/ec2: Explicitly convert bools and ints to strings when comparing policies

See also: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html

goneri committed:
VMware: not ssl.SSLContext if validate_certs false (#57185)

Python < 2.7.9 does not have the ssl.SSLContext attribute.
ssl.SSLContext is only required when we want to validate the SSL
connection. If `validate_certs` is false, we don't initialize the
`ssl_context` variable.

Add unit-test coverage and a little refactoring:

- avoid the use of `mocker`, when we can push `monkeypatch` which is
  `pytest`'s default.
- use `mock.Mocker()` when possible

closes: #57072

sivel committed:
Include version_added in report.py (#61857)

tdekeyser committed:
New module for AWS Step Functions state machines (#59116)

* add new module: aws_stepfunctions_state_machine

* add integration tests for new module: aws_stepfunctions_state_machine

* fix sanity checks

* use files/ folder instead for integration test

* rename role name in integration test

* attempt further permissions

* iam states prefix

* iam integration test prefix

* add iam policy for running step functions state machine actions

* slightly increase iam permission scope

* rename integration test folder to proper name

* move main() method to end of file

* move contents of integration-policy.json for state machines to compute-policy.json

* make check_mode return proper changed value + add check_mode integration tests

* rename module to aws_step_functions_state_machine

* fix missed rename in integration test variable

* add purge_tags option

* bump to version 2.10

goneri committed:
Revert "shippable: temporary bump the timeout for Azure jobs"

The Azure timeout should not accure anymore since
173d47d1f429847ca351da92b69a0e05d25bb313.

This reverts commit aaa8835311f39db42843efe19bd07c4691009eef.

jillr committed:
Fix ec2_vpc_vgw broken tests (#60981)

* Fix ec2_vpc_vgw broken tests

Add waiter function to wait for API to report detached vgw is available.
Also catch extra error code in attach retry as EC2 sometimes reports that
the vgw is available several seconds before permitting the attachment.

Fixes: #53185

* Re-enable ec2_vgc_vgw test target

jborean93 committed:
win_become - don't dispose logon token until end (#61905)

Qalthos committed:
Allow the use of _paramiko_conn even if the connection hasn't been started. (#61570)

* Allow the use of _paramiko_conn even if the connection hasn't been started.

I'm not sure what the benefit is of Noneing paramiko_conn on close, but will keep for now

* Fix test

* Try to fix up net_put & net_get

* Add changelog

tremble committed:
Fixup aws_secret integration tests (#61241)

* aws_secret: (integration tests) Move tests to using module_defaults

* Update hacking aws security policy to enable management of secrets

* aws_secret: (integration tests) Fixup integration tests
- Update tests to use resource_prefix as a prefix rather than a suffix
- Pause after role creation to cope with AWS being slow (and returning before the role it ready)

abadger committed:
create-deprecated-issues script can now add to a specified project (#61901)

* create-deprecated-issues script can now add to a specified project

* Migrate the create deprecated issues script into a subcommand of build-ansible

* Remove deprecated-issue script from ignore list

createchange committed:
Update playbooks_prompts.rst (#62003)

Fixed typo: "wierd" > "weird"

mattclay committed:
Fix issues with ansible-test --venv option. (#62033)

* Fix ansible-test venv activation.

When using the ansible-test --venv option, an execv wrapper for each python interpreter is now used instead of a symbolic link.

* Fix ansible-test execv wrapper generation.

Use the currently running Python interpreter for the shebang in the execv wrapper instead of the selected interpreter.

This allows the wrapper to work when the selected interpreter is a script instead of a binary.

* Fix ansible-test sanity requirements install.

When running sanity tests on multiple Python versions, install requirements for all versions used instead of only the default version.

* Fix ansible-test --venv when installed.

When running ansible-test from an install, the --venv delegation option needs to make sure the ansible-test code is available in the created virtual environment.

Exposing system site packages does not work because the virtual environment may be for a different Python version than the one on which ansible-test is installed.

felixfontein committed:
openssh_keypair: make sure public key has same permissions as private key (#61658)

* Make sure public key has same permissions as private key.

* Add changelog.

* Text, not binary.

mmyyrroonn committed:
fix ci (#57216)

mattclay committed:
Fix ansible-doc traceback and sanity test. (#62040)

* Fix ansible-doc traceback for removed modules.

This avoids tracebacks with errors like the following when a module has been removed:

module module_name missing documentation (or could not parse documentation): 'NoneType' object does not support item assignment

* Fix ansible-doc sanity test warning handling.

Warnings about removed modules/plugins on stderr are now properly ignored.

Previously an ansible-doc error could result in unrelated errors going undetected because tests were stopped early and the underlying error was ignored.

sagarafr committed:
fix(azure_rm_containerregistry): Fixed sku idempostency (#61009)

JKDingwall committed:
issue #61672: make jenkins_plugin module work in a session when CSRF … (#61673)

* issue #61672: make jenkins_plugin module work in a session when CSRF enabled

This commit modifies the signature of `fetch_url` so that a cookie jar can be
specified allowing multiple calls to operate with the same session.  It uses
a similar construct to the `Request` class to initialise the cookie jar if
it is not provided.

The jenkins_plugin module is modified to create a cookie jar if CSRF is
enabled.  This cookie jar is then submitted with every call to fetch_url.
Also changed is to submit the crumb in the request headers rather than
in the data field.

This has been tested with Jenkins 2.176.

* issue #61672: fix jenkins_script module

This commit modifies the jenkins_script module to use the authorization crumb
in a session in a similar fashion to the jenkins_plugin change for the same
issue.

Andersson007 committed:
postgresql: fix typos in modules (#62065)

Andersson007 committed:
mysql_info: fix typo (#62067)

decentral1se committed:
Update my handle (#61966)

chashtag committed:
Update ovirt_vm.py (#62072)

Typo fix: "IDE" is not an acceptable choice for disk type, "ide" is a valid choice

Akasurde committed:
mount: Check if src exists before mounted (#61752)

Fixes: #59183

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

mattclay committed:
Properly exclude tests/output/ from code coverage.

goneri committed:
bump hcloud version to 1.4.1 (#62097)

* bump hcloud version to 1.4.1

`hcloud`<=1.4.0 has requirement `requests==2.20.0`. This prevents the
installation of the Vcenter Automation SDK which depends on `requests>=2.22.0`.

`hcloud` 1.4.1 does not have the problem: https://github.com/hetznercloud/hcloud-python/commit/8bff356efb17f45458519f73beab5d8b41a79b8e
Bumping the dependency will resolve the issue.

mattclay committed:
Fix ansible-test to ignore `tests/output/`. (#62084)

The `test/results/` directory for Ansible test output was already ignored when not using git.

When Ansible Collections were switched to `tests/output/` the ignore entry was previously overlooked.

jillr committed:
AWS ec2_vpc_net: Enable ipv6 CIDR assignment (#60983)

* AWS ec2_vpc_net: Enable ipv6 CIDR assignment

Enable IPv6 CIDRs in ec2_vpc_net, and fix ec2_vpc_subnet tests that
were depending on the aws cli for CIDR assignment.

Related to: #27800

pixiake committed:
hostname: Add support for Kylin linux (#62053)

* Add support for Kylin linux to hostname module

mattclay committed:
Work around virtualenv/venv issue in ansible-test. (#62111)

Creating a virtual environment using `venv` when running in a virtual environment created by `virtualenv` results in a copy of the original virtual environment instead of creation of a new one.

To work around this, `ansible-test` now identifies when it is running in a `virtualenv` created virtual environment and uses the real Python interpreter to create the `venv` virtual environment.

ezombie committed:
hostname: Add support for Astra Linux Common Edition (#59607)

Add support Astra Linux Common Edition in hostname module

Akasurde committed:
hostname: Use hostnamectl for Systemd strategy (#59974)

Use hostnamectl command to get current hostname for host while using
systemd strategy.

Fixes: #59438

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

jborean93 committed:
Fix ansible-test coverage --all (#62115)

ganeshrn committed:
Return commands key instead of xml in result for junos rm (#62041)

Fixes https://github.com/ansible/ansible/issues/61773

*  Change `xml` key name to `commands` key to be in sync with
   other platform resource modules.

Im0 committed:
lookup_rabbitmq pika > 1.0.0 is_closing bug fix (#61959)

* In pika v1.0.0 BlockingChannel.is_closing was removed.  Updating
plugin accordingly.

Ref: https://github.com/pika/pika/pull/1034

* Adding change fragment for is_closing bug.

* Updated change fragment description.

mattclay committed:
Fix ansible-test pytest plugin loading. (#62119)

* Avoid assertion rewriting in pytest plugins.

Adding PYTEST_DONT_REWRITE to the ansible-test pytest plugin docstrings disables assertion rewriting in pytest for those plugins.

This avoids warnings during test execution if the plugins are loaded multiple times (such as being imported within tests).

* Run ansible-test pytest plugins early.

The ansible-test pytest plugins need to load and run earlier than conftest modules.

To facilitate this, the pytest_configure function is run during loading, which works since they are loaded (but not always run) before conftest modules are loaded.

A check has also been added to the pytest_configure functions to prevent them from running multiple times in the same process.

* Load pytest plugins using an env var.

The -p command line option loads plugins before conftest, but only during collection.
The PYTEST_PLUGINS environment variable loads plugins before confest, both during collection and test execution.

liviozanol committed:
fix regex in plugins/terminal/iosxr.py (#54817)

Fixes situations where iosxr terminals that do not contain new line "/r/n" at the beginning of CLI timeouts due to regex error. Just make "/r/n" optional including "*" character in the regex

rubentsirunyan committed:
New module: zabbix_valuemap (#60132)

* Adding zabbix_valuemap module

* Minor corrections

* Fixing typos

Co-Authored-By: Dusan Matejka <D3DeFi@users.noreply.github.com>

* Sorting mappings based on the 'value' field

* Updating

Co-Authored-By: John R Barker <john@johnrbarker.com>

* Fixing "version_added"

Andersson007 committed:
Fix typos in database modules and their integration tests (#62125)

kbreit committed:
Elaborate on note that docker-priv only works with integration (#62035)

danielmellado committed:
Fix test_cp_mgmt flake8 issues (#62129)

This commit deletes unused imports for the check_point cp_mgmt tests.

pabelanger committed:
Clean up more linting errors for nxos modules (#62069)

This removes a few unused variables and duplicate functions being
defined.

Signed-off-by: Paul Belanger <pabelanger@redhat.com>

mikewiebe committed:
Call get_capabilities to initiate device connection (#62103)

mikewiebe committed:
Add proper hostname cleanup for nxos tests (#61810)

heuels committed:
routeros_facts: fix for error when there's more than 10 interfaces (#61376)

* fix: proper regex for preprocessing routeros output

* test: regression test

* test: fix nondeterministic unit test

minievg committed:
Several enhancements for Radware modules and their unit testing (#61467)

* Adding support for Plugin runnable type
Adding support for device arrays in vdirect_runnable module.
Adding "output" dictionary to the vdirect_runnable module result dictionary.

* Adding support for Plugin runnable type
Adding support for device arrays in vdirect_runnable module.
Adding "output" dictionary to the vdirect_runnable module result dictionary.

sky-joker committed:
Added zabbix module zabbix_user (#56815)

This pull request adds a module to manage Zabbix users.
Tested with Zabbix Server(3.0, 4.0, 4.2).

ssbarnea committed:
docs: avoid confusing double negation (#62143)

Avoid "no backward incompatible interface" term which uses a double
negation and replaces it with easier "backward compatible interface"
contruct.

AlanCoding committed:
Allow tower inventory plugin to accept integer inventory_id (#61338)

axc450 committed:
Added AWS SAML Federation Module (#55821)

* Added AWS SAML Federation Module

* iam_saml_federation: (integration tests) initial integration tests

Im0 committed:
rabbitmq_publish fix for incorrectly stating message was not published to the queue (#61960)

* Pika v1.0.0 and above were causing issues for publish_message.  Updated
to ensure publish_message works with pika 0.13.1 and 1.0.0 and above.

* Adding changelog fragment for rabbitmq_publish fix.

* Updating return value.

Andersson007 committed:
fix typos in modules (#62135)

mulatinho committed:
luks_device.py: allows user explicity define luks format version (#61812)

* luks_device.py: allow the user create LUKS based on specific versions

- Allow user pass an option 'type' that explicits define the version of LUKS
  container that will be created. It should be 'luks1' or 'luks2' format.
- If 'label' option is defined the 'type' option will be 'luks2' independently
  of the option 'type' informed by user. (labels NEED luks2 format)

Fixes: #58973
Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

* added the changelog fragment

Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

* luks_device.py: make it fail in certain conditions

- Not allow user especify luks1 type and label at the same playbook

Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

Andersson007 committed:
hall: remove the module for supporting nonexistent messenger (#62152)

mattclay committed:
Add missing changelog entry for ansible-test fix.

PR https://github.com/ansible/ansible/pull/62119 was missing a changelog entry.

mattclay committed:
Remove deleted file from sanity ignore list.

felixfontein committed:
Docs formatting: recursively preprocess suboptions (#61900)

* Simplify code, move option massaging code into own function.
* Process suboptions recursively.

mattclay committed:
Fix ansible-test vcenter test filtering.

The default behavior of the ansible-test vcenter plugin is to use the govcsim container to run tests.

However, unless the govcsim mode was specified using the VMWARE_TEST_PLATFORM environment variable, the filter code would skip the tests unless the tests ran on Shippable or the user had an ansible-core-ci key.

Now the filter correctly recognizes that govcsim is the default.

mattclay committed:
Fix ansible-test collections requirements installation. (#62181)

* Fix location of unit test requirements.

* Preserve ansible-test unit test requirements.

* Remove redundant unit test requirements.

* Fix location of network test requirements.

* Preserve ansible-test network test requirements.

* Remove redundant network test requirements.

* Add missing ordereddict requirements.

* Load collection requirements correctly.

* Add changelog fragment.

NilashishC committed:
Set no_log to True for junos_user encrypted_password (#62184)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

NilashishC committed:
Handle cases where normal commit operation throws a prompt (#62132)

Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

Andersson007 committed:
elasticache_info: fix a typo (#62204)

Andersson007 committed:
fix typos in clustering modules (#62196)

Andersson007 committed:
fix typos in net_tools_modules (#62201)

dangothemango committed:
added version support to aws_ssm lookup (#61045)

Andersson007 committed:
fix typos in cloud modules (#62194)

* fix typos in cloud modules

* fix typos in cloud modules, fix ci tests

willthames committed:
Add exception handling when retrieving k8s client (#60726)

Andersson007 committed:
svc: fix a typo (#62174)

Andersson007 committed:
fix typos in web_infrastructure modules (#62202)

Andersson007 committed:
fix typos in identity modules (#62200)

Andersson007 committed:
fix typos in commands modules (#62197)

zombah committed:
Don't fail if exos can't configure cli columns (#51364)

* Don't fail if exos can't configure cli columns

Signed-off-by: Misha Komarovskiy <zombah@gmail.com>

bcoca committed:
Revert "svc: fix a typo (#62174)" (#62212)

This reverts commit d838a9021a8f7e951bc5dc5dd40fd5686f58a8c1.

was not a typo, 'ed' is added later on and 'stop' goes to 'stopped'

chrisvanheuveln committed:
nxos_lacp_interfaces: fix integration test dependencies (#61947)

* nxos_lacp_interfaces: fix integration test dependencies

* Add conditional mode test to m/d/r tests

mikewiebe committed:
Fix bfd cmd order and test issues (#61943)

mikewiebe committed:
[stable-2.9] fix nxos_config tests for httpapi (#62082)

* fix nxos_config tests for httpapi

* Remove become parameter

chrisvanheuveln committed:
nxos_l2_interfaces: fix for integration tests failing to setup layer2 (#61887)

abadger committed:
Add guidelines for when to use a doc_fragment (#61828)

samdoran committed:
Improve documentation on doc fragments

Add information and examples on how to use additional properites from a doc fragment

samdoran committed:
Remove suggestion to go caving

akira6592 committed:
explicited RouterOS does not support connection: local (#62203)

ctrufan committed:
Always specify header of connection keep-alive regardless of python v… (#62218)

* Always specify header of connection keep-alive regardless of python version.

* Add chgangelog fragment

* Fixes to changelog fragment

geerlingguy committed:
Fixes #56832: Remove warning when falling back to apt-get if aptitude is missing. (#61782)

Akasurde committed:
core: Handle empty extra vars in cli (#61831)

Fixes: #61497

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Commit:
fix fact caching plugin name from json to jsonfile (#62198)

evgeni committed:
typo: missing-subption-docs → missing-suboption-docs (#62180)

updates the name of the test and all ignore.txt entries

evgeni committed:
sanity ignores for a collection need to go to tests/sanity/ (#62182)

collections use tests/ not test/ as the folder for tests

evgeni committed:
document suboptions for type:list options too (#62177)

felixfontein committed:
openssh_keypair file permissions/ownership: add porting guide entry (#62176)

* Add porting guide entry for 2.9.

s-hertel committed:
Add some details to any_errors_fatal documentation (#62029)

jtyr committed:
Updating link for guest_id info (#61988)

samdoran committed:
Add info about layering properties

samdoran committed:
nxos_bfd_global - add missing import of re (#62226)

jborean93 committed:
ansible-galaxy - add config to control the display wheel (#61902)

* ansible-galaxy - add config to control the display wheel

* Fix changelog and make test more stable

* Don't use display thread at all if progress wheel isn't being shown

Akasurde committed:
VMware: Change return key `results` from VMware modules (#62161)

Following module used internal results key as part of return json, this commit
changes this to appropriate values -
* vmware_datastore_maintenancemode.py
* vmware_host_kernel_manager.py
* vmware_host_ntp.py
* vmware_host_service_manager.py
* vmware_tag.py

Fixes: #62083

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

termlen0 committed:
Update junos_config.py (#62221)

Add the commands alias to match UX of IOS, EOS etc.

Commit:
Update BOTMETA.yml (#62123)

kr3ator committed:
Update junos_user.py (#60459)

chkp-orso committed:
Add unit tests for check_point ansible modules (#62213)

* Update test_cp_mgmt_network.py

* 17 tests

s-hertel committed:
remove undefined YAML anchor (#62273)

Glandos committed:
Update mitogen link (#62247)

ganeshrn committed:
Fix for junos cli_config replace option (#62131)

* Fix for junos cli_config replace option

*  For device that support replace option by loading
   configuration from a file on device `config` option
   is not required and value of `replace` option is the
   path of configuration file on device. This fix allows
   invoking run() function in cli_config if `config` option
   is None and `replace` option is not boolean

*  The command to replace running config on junos device
   is `load override <filename>` and not `load replace <filename>`
   This is fixed in the junos cliconf plugin.

* Add integration test

Qalthos committed:
Standardize eos resource modules (#61736)

* Fix eos_l3_interfaces case sensitivity

* Unify EOS module notes

* Add normalize_interfaces to eos_l2_interfaces

* Pull normalize_interface into eos_interfaces

* Add normalize_interface to lag_interfaces

* Add normalize_interface to lldp_interfaces

* Add normalize_interface to lacp_interfaces

* more module cleanup

* Add changelog

felixfontein committed:
Re-enable some docker_swarm tests. (#61875)

samdoran committed:
Update default test container with Python 3.8b4 (#62100)

sivel committed:
Don't treat no checksum as a checksum match (#62146)

Fixes #61978
* moar tests for get_url fetch behavior with existing file
* add changelog fragment

sivel committed:
Ensure we don't erase unsafe context in TaskExecutor.run on bytes (#62287)

* Ensure we don't erase unsafe context in TaskExecutor.run on bytes. Fixes #62237

* Remove unused import

* Add missing import

* use args splatting for to_unsafe_text/bytes

* Add security issue to changelog

* fix yaml linting issue

n3pjk committed:
Add httpapi VMware REST VmwareRestModule (#60914)

* Initial commit of VMware HttpApi REST

New directory structure created for httpapi-based modules. These will
live in the 'vmware_httpapi' directory under 'modules/cloud'.  The
AnsibleModule class was extended to create the VmwareRestModule class
that lives in the 'vmware_httpapi' directory under 'module_utils'. It
implements comms with the httpapi plugin, and also provides debugging
output, url and filtering generation by object, and support for
multiple VMware REST APIs. It also provides dynamic handling of HTTP
return codes that can be tailored to each module's needs.

flowerysong committed:
Add version 2.14 to the removed_in list (#62298)

devel is now 2.10, so this needs to be bumped as well.

samccann committed:
revert back to contiguous (#62301)

goneri committed:
azure: re-balance tests to avoid timeout

- azure_rm_azurefirewall currently takes more than 25m to run: #62307
- azure_rm_manageddisk takes around 2 minutes

mattclay committed:
Fix display of sanity test doc links.

The documentation links are now displayed when running from an install.

Previously the links were only displayed when running from source.
This was due to ansible-test checking for the presence of documentation files locally, which are only present when running from source.
The check is no longer necessary since there is a sanity test in place to enforce the presence of documentation for all sanity tests.

chkp-orso committed:
Add unit tests for check_point ansible modules (#62214)

* Update test_cp_mgmt_network.py

* 18 tests

chkp-orso committed:
Third pr 18 tests (#62215)

* Update test_cp_mgmt_network.py

* 18 tests

ctrufan committed:
Add ecs_domain module (#62007)

* Add ecs_domain module

* Fixes to integration tests and module

* Fixes to tests and module

* Corrections to revalidation behavior, cna only revalidate domains in expiring.

* Remove debugs for final test run, fix sanity check test fails.

* Add checks for domain status

* Add changelog fragment for new module.

* Removed extra space in backtick

* Minor fixes to make behavior more consistent and correct documentation.

* Update lib/ansible/modules/crypto/entrust/ecs_domain.py

Co-Authored-By: Felix Fontein <felix@fontein.de>

* Update lib/ansible/modules/crypto/entrust/ecs_domain.py

Co-Authored-By: Felix Fontein <felix@fontein.de>

* Update lib/ansible/modules/crypto/entrust/ecs_domain.py

Co-Authored-By: Felix Fontein <felix@fontein.de>

* Apply suggestions from code review

Co-Authored-By: Felix Fontein <felix@fontein.de>

* Change casing of verification method enum, remove redundant changelog fragment

* Return ov_eligible and ev_eligible fields even if false, as long as they're returned by ECS API

felixfontein committed:
Remove deprecated letsencrypt alias of acme_certificate (#61648)

briantist committed:
Fix check mode support in win_group (#61977)

* Fix check mode support in win_group

* Fix copy/paste error

joshinryz committed:
Support large folder size comparisons for win_find (#58466)

* Support large folder size comparisons for win_find

Changed [int] to [int64] to support larger folders. Otherwise module fails as soon as a large folder is encountered.

* Create 58466-FIX_win_find-Bug-Get-FileStat_fails_on_large_files.yml

ShachafGoldstein committed:
win_find - Fix deduped files mistaken for directories (#58680)

* win_find - Fix deduped files mistaken for directories

* Update win_find.ps1

* Update win_find.py

* Update win_find.ps1

jn-bedag committed:
correct behaviour of win_iis_webapplication when no applicationpool is specified (#61227)

* correct behaviour of win_iis_webapplication when no applicationpool is specified

* documentation for win_iis_webapplication apppool change

Fred-sun committed:
add paramter type

Fred-sun commented:
@yungezz @MyronFanQiu Could you please help to cancel my submission this time? Thank you very much!
